tests/apps/manager_app/manager_api_test.c

```diff
@@ -0,0 +1,158 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+#include "manager_api.h"
+
+#include <stdbool.h>
+#include <string.h>
+
+#include "unity_fixture.h"
+
+TEST_GROUP(manager_api_test);
+
+/**
+ * @brief Test setup for usb event consumer tests.
+ * @details The function populates data in local buffer of USB communication
+ * module so that the event getter has an event ready for dispatch for
+ * performing tests. buffer of packet(s) of data.
+ */
+TEST_SETUP(manager_api_test) {
+}
+
+/**
+ * @brief Tear down the old test data
+ * @details The function will perform cleanup of the current running test and
+ * bring the state of execution to a fresh start. This is done by using purge
+ * api of usb-event and clearing buffers using usb-comm APIs.
+ */
+TEST_TEAR_DOWN(manager_api_test) {
+}
+
+TEST(manager_api_test, decode_valid_manager_bs) {
+  manager_query_t query_out = {0};
+  uint8_t byte_stream[] = {34,  36,  18,  34,  10,  32,  6,   4,   37,  226,
+                           45,  95,  166, 129, 63,  247, 201, 188, 151, 71,
+                           113, 179, 160, 120, 135, 252, 54,  30,  198, 28,
+                           152, 73,  180, 129, 55,  26,  238, 246};
+  uint8_t challenge[] = {
+      0x06, 0x04, 0x25, 0xe2, 0x2d, 0x5f, 0xa6, 0x81, 0x3f, 0xf7, 0xc9,
+      0xbc, 0x97, 0x47, 0x71, 0xb3, 0xa0, 0x78, 0x87, 0xfc, 0x36, 0x1e,
+      0xc6, 0x1c, 0x98, 0x49, 0xb4, 0x81, 0x37, 0x1a, 0xee, 0xf6,
+  };
```

Comment:
could we use encoder to prepare test data?

Reply:
can be as a separate change request

Reply:
Currently we have created apis for decode or query and encoding result, there will be requirement of additional API here. As we plan on creating a common API for this, we can take it up as a follow up change.